### DISCUSSION TITLE
How to rotate a page

### DISCUSSION THREAD

**Discussion by stevecoh1** [xournalpp/xournalpp#5472] [Q&A]
I have the simplest use case but I can't figure out how to do it. I have a pdf document that is inverted 180 degrees and I merely want to rotate it.  I see that there is a "rotation snapping" tool which I can turn on and off, but I don't see any control that would allow me to actually rotate the page or the selection or anything.  What am I missing? Version is 1.2.1.  Platform is Ubuntu Linux (23.10).

**Comment by netlimpopo** [comment]
Hello!  Unfortunately there is no such feature in Xournal++ yet. One can only move a pdf page, or delete them. As Xournal++ works on top of the PDF document and keeps the original file AND the .xopp file, one can change the PDF with an external program. Its best to apply changes to the PDF first and then start to annotate on the file in Xournal++. 
As long as one does not change the order nor does not delete pages in the orignal PDF chages can be done at any point, as the page numbers won't be affected. So rotating pages should be fine at any time.

**Comment by yzsshen** [comment]
Being able to rotate pages or the window as a whole would be a great feature.

**Comment by Nick-b404** [comment]
This still doesn't work and it's a rather basic feature of a PDF editor. There seems to be a 'Paper format' option under 'Journal' but it's greyed out. How come this still doesn't work?

**Comment by Sylvain303** [comment]
Just sharing and save you some click, 🎁

As a workaround you can rotate the PDF outside xournal and then edit it

search engine gave me:

> Command line How do you rotate a PDF file 90 degrees
> http://unix.stackexchange.com/questions/394065/ddg#394068
> 
> Use PDFtk.

    For rotating clockwise:

     pdftk input.pdf cat 1-endeast output output.pdf

    For rotating anti-clockwise:

     pdftk input.pdf cat 1-endwest output output.pdf

**Comment by JMak22** [comment]
Another possibility for rotating PDFs is via Inkscape.
Import PDF into Inkscape, select each page you wish to rotate, click on the page again, while selected, use one of the rounded corner arrow keys to rotate the page.
File-->Export as PDF

If you have access to Adobe Document Cloud, you can rotate pages there as well, but it may be a paid feature.

**Comment by Moini** [comment]
Is there a way to turn this into a feature request? The Q&A category doesn't do it justice.  Workarounds are nice, but ... the page's rotation could be saved to the .xopp file along with all the other things. I often work through documents that contain tables which are too large for being printed upright, so they are printed sideways. I can't read them in Xournal++.

**Reply by The-An0nym** [reply]
Feature requests happen via issues as far as I'm aware.
See https://github.com/xournalpp/xournalpp/issues/1883

I've taken an interest in implementing such a feature but cannot make any promises as of right now.

**Reply by Moini** [reply]
Ah, that's cool - thank you, @The-An0nym ! (and good to have the cross link now, I think I forgot to do a proper search :flushed: )
